README.md
# MapirAngularComponent
[![npm version](https://badge.fury.io/js/mapir-angular-component.svg)](https://www.npmjs.com/package/mapir-angular-component)

![Map.ir](https://map.ir/css/images/mapir-logo.png) Angular wrapper for mapbox-gl-js. Expose a bunch of component meant to be simple to use for Angular.

## Get API Key
🔑 You should first get api key from [Map.ir](https://corp.map.ir/registration/)

## Installation

```bash
npm i mapir-angular-component mapbox-gl
```

If using typescript, add mapbox-gl types:

```bash
npm install @types/mapbox-gl
```

### Add mapbox-gl CSS

Load the css of mapbox-gl (and mapbox-gl-geocoder if mglGeocoder is used)

For example, with angular-cli projects add this in `angular.json`:

```json
"styles": [
  "./node_modules/mapbox-gl/dist/mapbox-gl.css",
  "./node_modules/@mapbox/mapbox-gl-geocoder/lib/mapbox-gl-geocoder.css"
  ]
```
Or in global css (called styles.css for example in angular-cli)

```css
@import "~mapbox-gl/dist/mapbox-gl.css";
@import "~@mapbox/mapbox-gl-geocoder/lib/mapbox-gl-geocoder.css";
```

### Add Polyfill

Add this in your polyfill.ts file (Wykks/ngx-mapbox-gl/issues/136#issuecomment-496224634):

```ts
(window as any).global = window;
```

## Quick start 

### Import module

in `app.module.ts` file:

```js
import { NgxMapboxGLModule } from 'mapir-angular-component';

@NgModule({
  imports: [
    NgxMapboxGLModule
  ]
})
export class AppModule {}
```

### Set API Key

in `app.component.ts` file:

```diff
@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'mapir-angular-test';
  center:Array<number> = [35.712706, 51.367918];
+  apiKey:string = "<YOUR API KEY>";
}
```

### Generate Component

in `app.component.html` file:

```html
<mgl-map
  [zoom]="[2]"
  [center]="center"
  [centerWithPanTo]="true"
  [interactive]="true"
  [apiKey]="apiKey"
></mgl-map>
```

[![Edit stackblitz](./assets/stackblitz.png)](https://stackblitz.com/edit/mapir-angular-component-test)

[![Edit mapir-angular-component-test](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/vibrant-sound-obw3p?fontsize=14&hidenavigation=1&theme=dark)

📖 [English Documentation](https://github.com/map-ir/mapir-angular-component/wiki/Documentation)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [4.2.1](https://github.com/map-ir/ngx-mapbox-gl/compare/v4.2.0...v4.2.1) (2019-12-07)

## [4.2.0](https://github.com/Wykks/ngx-mapbox-gl/compare/v4.1.0...v4.2.0) (2019-11-19)


### Features

* **mglGeolocate:** add 'geolocate' event output to component ([#193](https://github.com/Wykks/ngx-mapbox-gl/issues/193)) ([adb9382](https://github.com/Wykks/ngx-mapbox-gl/commit/adb93821483e2708b971bf7da972024afdcebd07))

## [4.1.0](https://github.com/Wykks/ngx-mapbox-gl/compare/v4.0.1...v4.1.0) (2019-10-23)


### Features

* add `clickTolerance` as an Input for MapComponent  [#135](https://github.com/Wykks/ngx-mapbox-gl/issues/135) ([#190](https://github.com/Wykks/ngx-mapbox-gl/issues/190)) ([055f90e](https://github.com/Wykks/ngx-mapbox-gl/commit/055f90e976b12816e71a5bf038df34f9f3af588d))
* allow using mapbox-gl >= 1.1.0 ([80cdfab](https://github.com/Wykks/ngx-mapbox-gl/commit/80cdfabd9bec122d548498a7ac6569b7139dfb00))


### Bug Fixes

* make sure showcase works with TypeScript 3.5.3 ([cbaa95c](https://github.com/Wykks/ngx-mapbox-gl/commit/cbaa95caf4740c3958a9aa1361bbad8f246e53ab))

### [4.0.1](https://github.com/Wykks/ngx-mapbox-gl/compare/v4.0.0...v4.0.1) (2019-10-13)


### Bug Fixes

* **popup:** actually remove the popup from the map when a "feature" is used as an anchor ([89f4aa4](https://github.com/Wykks/ngx-mapbox-gl/commit/89f4aa4))



# [4.0.0](https://github.com/Wykks/ngx-mapbox-gl/compare/v3.3.0...v4.0.0) (2019-10-11)


### Bug Fixes

* **controls:** Fix control order when using custom controls alongside built-in controls (fix [#75](https://github.com/Wykks/ngx-mapbox-gl/issues/75)) ([10612f0](https://github.com/Wykks/ngx-mapbox-gl/commit/10612f0))
* **geocoder:** change property name for country option to countries ([781adc4](https://github.com/Wykks/ngx-mapbox-gl/commit/781adc4))


### Features

* update to angular 8 & other deps ([57f6d6e](https://github.com/Wykks/ngx-mapbox-gl/commit/57f6d6e))
* update to mapbox-gl 1.1.0 ([1043b4f](https://github.com/Wykks/ngx-mapbox-gl/commit/1043b4f))
* **marker:** Add ability to specify className (close [#172](https://github.com/Wykks/ngx-mapbox-gl/issues/172)) ([f7602c2](https://github.com/Wykks/ngx-mapbox-gl/commit/f7602c2))
* **popup:** Add support for className ([#111](https://github.com/Wykks/ngx-mapbox-gl/issues/111)) ([#161](https://github.com/Wykks/ngx-mapbox-gl/issues/161)) ([3b30c90](https://github.com/Wykks/ngx-mapbox-gl/commit/3b30c90))
* **popup:** Add support for maxWidth ([#180](https://github.com/Wykks/ngx-mapbox-gl/issues/180)) ([04d910b](https://github.com/Wykks/ngx-mapbox-gl/commit/04d910b))


### Refactor

* **marker:** remove old css hack (now fixed in mapbox-gl) (close [#114](https://github.com/Wykks/ngx-mapbox-gl/issues/114)) ([9864932](https://github.com/Wykks/ngx-mapbox-gl/commit/9864932))


### BREAKING CHANGES

* Update to angular 8. Drop support of Angular 7
Also drop support of rxjs < 6.5.0
* Since mapbox-gl 1.0, there's a new pricing model.
If you use mapbox services, please check https://github.com/mapbox/mapbox-gl-js/releases/tag/v1.0.0 for more info
* **marker:** You may be affected by this change, if you use html markers with custom css.


# [3.3.0](https://github.com/Wykks/ngx-mapbox-gl/compare/v3.2.0...v3.3.0) (2019-07-03)

### Bug Fixes

* **image:** Handle coordinate update correctly (close [#139](https://github.com/Wykks/ngx-mapbox-gl/issues/139))

### Features

* **controls:** added input for customAttributions to attribution-control directive ([c93e118](https://github.com/Wykks/ngx-mapbox-gl/commit/c93e118))
* **map:** support antialias option (close [#148](https://github.com/Wykks/ngx-mapbox-gl/issues/148)) ([b03d8f8](https://github.com/Wykks/ngx-mapbox-gl/commit/b03d8f8))
* **map:** support styleimagemissing event ([056b61f](https://github.com/Wykks/ngx-mapbox-gl/commit/056b61f))
* update mapbox-gl to 0.54.0 ([fcc0a58](https://github.com/Wykks/ngx-mapbox-gl/commit/fcc0a58))


# [3.2.0](https://github.com/Wykks/ngx-mapbox-gl/compare/v3.1.0...v3.2.0) (2019-06-08)

### Bug Fixes

* **map:** Add safety check for MapService.mapInstance before attempting to remove (close [#142](https://github.com/Wykks/ngx-mapbox-gl/issues/142)) ([cddeb2af](https://github.com/Wykks/ngx-mapbox-gl/commit/cddeb2af))

### Features

* expose MAPBOX_GEOCODER_API_KEY (close [#152](https://github.com/Wykks/ngx-mapbox-gl/issues/152)) ([67be0b4](https://github.com/Wykks/ngx-mapbox-gl/commit/67be0b4))
* **markers-for-cluster:** new component markers-for-cluster ([a5b97f6](https://github.com/Wykks/ngx-mapbox-gl/commit/a5b97f6))
* **geojson-source**: expose getClusterExpansionZoom getClusterChildren getClusterLeaves functions  ([40bf0852](https://github.com/Wykks/ngx-mapbox-gl/commit/40bf0852))

Deprecate mgl-marker-cluster



# [3.1.0](https://github.com/Wykks/ngx-mapbox-gl/compare/v3.0.1...v3.1.0) (2019-04-07)


### Bug Fixes

* **map:** exposing error event data to directive output. ([#127](https://github.com/Wykks/ngx-mapbox-gl/issues/127)) ([a611858](https://github.com/Wykks/ngx-mapbox-gl/commit/a611858))


### Features

* **map:** add bounds input (new as mapbox-gl 0.53.0) ([157b28f](https://github.com/Wykks/ngx-mapbox-gl/commit/157b28f))
* update to mapbox-gl 0.53.1 and fix compilation errors ([#128](https://github.com/Wykks/ngx-mapbox-gl/issues/128)) ([ed6634c](https://github.com/Wykks/ngx-mapbox-gl/commit/ed6634c))
* **marker:** Add popupShown Input binding to marker.component ([#113](https://github.com/Wykks/ngx-mapbox-gl/issues/113)) ([1b0fed8](https://github.com/Wykks/ngx-mapbox-gl/commit/1b0fed8))



<a name="3.0.1"></a>
## [3.0.1](https://github.com/Wykks/ngx-mapbox-gl/compare/v3.0.0...v3.0.1) (2019-01-27)


### Bug Fixes

* correct package.json ([4a42008](https://github.com/Wykks/ngx-mapbox-gl/commit/4a42008))



<a name="3.0.0"></a>

projects/simple/src/app/app.component.ts
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'mapir-angular-test';
  center:Array<number> = [51.367918, 35.712706];
  apiKey:string = "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImp0aSI6IjM0NzU5ZjVlOTZiZDc1YjNkMWRmZTA5M2JjNmNkYWZmY2ExMTg4ZDMwMTRmOWYzZWFiZjQyMTIyMzNlOTIzMzAzMjgwNGNjZDMxZjM4MTE3In0.eyJhdWQiOiIxMTQiLCJqdGkiOiIzNDc1OWY1ZTk2YmQ3NWIzZDFkZmUwOTNiYzZjZGFmZmNhMTE4OGQzMDE0ZjlmM2VhYmY0MjEyMjMzZTkyMzMwMzI4MDRjY2QzMWYzODExNyIsImlhdCI6MTU2MjQ5ODk5NiwibmJmIjoxNTYyNDk4OTk2LCJleHAiOjE1NjUxNzczOTIsInN1YiI6IiIsInNjb3BlcyI6WyJiYXNpYyJdfQ.ZKWm-NmJ9pI5eg33S5LQ2YxrpPmEfbnDm_IKXNxc-eWeYV22nayyhuvGT4qyd5Ao-E3_NcA9sl7NP5LD0gPBl6Efz2q_AsDkBPtQr8u22K3c7HSfS9msSsjHO3uKknehiTZ-agpYNoAiPWfk49DeUagw0w2EAJq3Y12pYC0yYeEYsB54s7zSFDOV8P2OqUUZS9JjF5OW4G0IsVHvyBrYCi0TRh45HallljvL3RPqmrvQUzkOYAY-3tw_zhbRgCWRdQ_tdt7D7JaK18xGrOPJfp2771YXvevsFQ39FvXZurFwcvNGrmRJxPTXyU-o92ptpkpRraI4PrFgtuTa5DYKbA";
}
